sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp

```diff
@@ -214,8 +217,45 @@ namespace Azure { namespace Security { namespace Attestation {
      */
     Response<Models::TpmAttestationResult> AttestTpm(
         AttestTpmOptions const& options,
-        Azure::Core::Context const& context = Azure::Core::Context::ApplicationContext) const;
+        Azure::Core::Context const& context = Azure::Core::Context{}) const;
+
+  private:
+    Azure::Core::Url m_endpoint;
+    std::string m_apiVersion;
+    std::shared_ptr<Azure::Core::Credentials::TokenCredential const> m_credentials;
+    std::shared_ptr<Azure::Core::Http::_internal::HttpPipeline> m_pipeline;
+    AttestationTokenValidationOptions m_tokenValidationOptions;
+
+    mutable std::vector<Models::AttestationSigner> m_attestationSigners;
```

Comment:
@CaseyCarter what's our guideline on using `mutable` vs avoiding it in private C++ members? I thought we'd avoid using `const` whenever we have APIs that mutate internal state, rather than using `mutable`.

@antkmsft, @LarryOsterman, @JeffreyRichter - what are your thoughts on this as well?

Reply:
A `mutable` member in a class without a comment to indicate how that member is accessed thread-safely from simultaneous calls to `const` member functions is a huge red flag.

Reply:
@CaseyCarter how do we guarantee (and document) thread-safety for other non-const methods, that do end up modifying the private members?
Does `mutable` introduce an extra "burden-of-proof"?

Reply:
Service client objects are supposed to be thread safe, which in practice means they need to be immutable. That in turn means that all service methods need to be marked as `const` (so they can be called from 'const` clients).

Having said that, the only method which modifies the `m_attestationSigners` field is the `RetrieveResponseValidationCollateral` and that is called from the factory on a non-const instance of the client. So `RetrieveResponseValidationCollateral` doesn't need to be marked as `const`.